### PR TITLE
Unify polls routes into controller

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #4902

## Description of Changes
- Move polls logic to new polls controller, methods and handlers
- Updates UI to use new routes
- Misc
  - Delete unused editComment route
  - Fixes some imports that used the wrong `ServerError` type

## Test Plan
- API tests were added for the new routes
- Manual CA tested
  - Create a new thread
  - On the thread, create a poll– confirm that UI shows new poll
  - Submit a vote– confirm UI updates state to show vote
  - Refresh the page– confirm UI shows poll and vote
  - Delete the poll– confirm UI shows poll is gone
  - Refresh the page again– confirm UI again shows that poll is gone

## Deployment Plan
N/A

## Other Considerations
- Have not yet removed the old routes, as I want to ensure the new implementations are correct first
- The UI doesn't allow updating a vote, but the API does (both old and new implementation)

### CODE REVIEW

**Comment by CowMuon** [comment]
Let's discuss getting the back half of the test plan into Playwright. Not to block this PR, and not to wait for our new QA Engineer to officially join the team on Sept 11; rather, aligning on a general approach, with an emphasis on reusability. Happy to discuss on W.E.M. (aka Woke Wednesdays.) @kurtisassad @rbennettcw @jnaviask @gdjohnson

**Review by CowMuon — COMMENTED** [review]
Let's have the Playwright discussion before merging this, if feasible.

**Comment by rbennettcw** [comment]
I've created unit tests and API tests before, but not Playwright tests.

@kurtisassad Any tips/guidelines for how to add a Playwright test to this PR?

**Comment by kurtisassad** [comment]
Yeah so basically we don't have any e2e tests for polls, so although this PR is just a cleanup, if you were to write e2e tests for playwright tests you basically need to code in the steps you have in your test plan. Playwright acts as a front end user, clicking around on things.

Some guidelines:
- write it in the e2eRegular/discussion.spec.ts, since this tests the /discussion route.
- You can follow other tests in this file for guidance.
- When adding locators (the code to find the buttons), try to specify them with a general property (some class name of the element)

Feel free to message me async if you have any other questions.

**Comment by rbennettcw** [comment]
Thanks for the breakdown– I'm working on adding an e2e test for polls.

**Comment by jnaviask** [comment]
@rbennettcw can you mark as draft until the test is ready? Thanks!

**Comment by rbennettcw** [comment]
I've got the e2e tests running locally, but adding a new test for polls is proving to be a non-trivial effort.
- Using the default seed data, the ViewThread page fails to load due to NaN thread ID
  - Updating the seed logic may break existing tests
- Using custom seed logic to create the thread fixes that issue, but then I need to reference an address/user to own the thread– and the existing `login` function doesn't allow specifying which user is logged in.

So it'll take more time to get a basic test working, and it'll likely take some bigger code changes to get multi-user tests working.

---

I'm putting this PR on hold for now so I can get a start on the Gating API– will revisit this next week. Maybe the new QA engineer can help take a look at this, since I'll be OOO Sept 11-13.

**Comment by kurtisassad** [comment]
Can you just create a thread through the e2e workflow (following the steps on the newDiscussion.spec.ts)? Then you will have a valid thread owned by the mocked metamask user.

**Comment by jnaviask** [comment]
@rbennettcw any reason this PR is still drafted? Can merge once we fix conflicts.

**Comment by rbennettcw** [comment]
Fixed.

FYI: although the old test was removed, there's a new test which has the same test coverage.

File: packages/commonwealth/server/controllers/server_polls_methods/delete_poll.ts
Comment (mzparacha) [inline packages/commonwealth/server/controllers/server_polls_methods/delete_poll.ts:28]: 'address' is defined but never used

File: packages/commonwealth/server/routes/threads/create_thread_poll_handler.ts
Comment (mzparacha) [inline packages/commonwealth/server/routes/threads/create_thread_poll_handler.ts:29]: left by mistake i think?

**Review by mzparacha — APPROVED** [review]
LGTM! 🚀.

**Review by Rotorsoft — APPROVED** [review]
Very similar to PR #5283. Noticed the extra effort documenting API changes - we can design a self-documented API, other than that LGTM 🚀

**Comment by timolegros** [comment]
This PR should be merged before work on https://github.com/hicommonwealth/commonwealth/issues/5357 begins.

**Pull request merged**
